common/frame_session.go

```diff
@@ -242,6 +242,11 @@ func (fs *FrameSession) initEvents() {
 			// If there is an active span for main frame,
 			// end it before exiting so it can be flushed
 			if fs.mainFrameSpan != nil {
+				// The url needs to be added here instead of at the start of the span
+				// because at the start of the span we don't know the correct url for
+				// the page we're navigating to. At the end of the span we do have this
+				// information.
+				fs.mainFrameSpan.SetAttributes(attribute.String("navigation.url", fs.manager.MainFrame().URL()))
```

Comment:
Can we add this (`fs.manager.MainFrame().URL()`) as a method on `FrameManager`? A train of calls is usually burdensome to maintain and work correctly.

Reply:
Sure, made the change in bf1dcbe280acd2543a55d08fbdacd90c5544628d